collectors/mimecast/utils.js

```diff
@@ -0,0 +1,247 @@
+
+            if (nextPage === undefined) {
+                payload = {
+                    "meta": {
+                        "pagination": {
+                            "pageSize": 100
+                        }
+                    },
+                    "data": [
+                        {
+                            "oldestFirst": true,
+                            "from": state.since,
+                            "route": "all",
+                            "to": state.until
+                        }
+                    ]
+                };
+            }
+            else {
+                payload = {
+                    "meta": {
+                        "pagination": {
+                            "pageSize": 100,
+                            "pageToken": nextPage
+                        }
+                    },
+                    "data": [
+                        {
+                            "oldestFirst": true,
+                            "from": state.since,
+                            "route": "all",
+                            "to": state.until
+                        }
+                    ]
+                };
+            }
+            break;
+        case Malware_Feed:
+            uri = `/api/ttp/threat-intel/get-feed`;
+            if (nextPage === undefined) {
+                payload = {
+                    "data": [
+                        {
+                            "fileType": "stix"
+                        }
+                    ]
+                };
+            }
+            else {
+                payload = {
+                    "data": [
+                        {
+                            "fileType": "stix",
+                            "token": nextPage
+                        }
+                    ]
+                };
+            }
+            break;
+        default:
+            uri = null;
+    }
+
+    return {
+        uri,
+        payload
+    };
+}
+
+function generateHeaders(authDetails, uri) {
+    let requestId = uuidv4().toString();
+    let hdrDate = moment().utc().format('ddd, DD MMM YYYY HH:mm:ss z');
+
+    let hmac = crypto.createHmac("sha1", Buffer.from(authDetails.secretKey, 'base64'));
+    hmac.write(`${hdrDate}:${requestId}:${uri}:${authDetails.appKey}`);
+    hmac.end();       // can't read from the stream until you call end()
+    signature = hmac.read().toString('base64'); 
+
+    return {
+        "Authorization": `MC ${authDetails.accessKey}:${signature}`,
+        "x-mc-app-id": authDetails.appId,
+        "x-mc-date": hdrDate,
+        "x-mc-req-id": requestId,
+        "Content-Type": 'application/json'
+    };
+}
+
+function getTypeIdAndTsPaths(stream) {
+    let typeIdPaths = [];
+    let tsPaths = [];
+
+    switch (stream) {
+        case Siem_Logs:
+            typeIdPaths = [{ path: ["aCode"] }];
+            tsPaths = [{ path: ["datetime"] }];
+            break;
+        case Attachment_Protect_Logs:
+        case URL_Protect_Logs:
```

Comment:
Do these two have same format?
In tests I see two examples: `CLICK_LOGS_EVENT ` and `LOG_EVENT ` which have different structure.
Could you please provide examples of logs for these two streams?